spec/services/metrics/project_review_turnaround_processor_spec.rb

```diff
@@ -0,0 +1,190 @@
+        pull_request_event_payload = create_pull_request_event(
+          action: 'opened',
+          created_at: Time.zone.parse('2020-01-01T15:10:00')
+        )
+
+        create_review_event pull_request_event_payload: pull_request_event_payload,
+                            action: 'submitted',
+                            submitted_at: Time.zone.parse('2020-01-01T15:30:00')
+
+        create_review_event pull_request_event_payload: pull_request_event_payload,
+                            action: 'submitted',
+                            submitted_at: Time.zone.parse('2020-01-01T16:30:00')
+      end
+
+      it 'it uses only the first review to calculate the metric value' do
+        expect(first_metric_value_expressed_as_seconds).to be_within(1.second) .of(20.minutes)
+      end
+    end
+  end
+
+  describe 'with a PR that has no reviews' do
+    let(:create_test_events) do
+      create_pull_request_event(action: 'opened',
+                                created_at: Time.zone.parse('2020-01-01T15:10:00'))
+    end
+
+    it 'does not generate a metric' do
+      expect(generated_metrics_count).to eq(0)
+    end
+  end
+
+  describe 'with events from more than one project' do
+    let(:test_repository_b_payload) do
+      (build :repository_payload, name: 'Project B')['repository']
+    end
+
+    let(:create_test_events) do
+      pull_request_a_event_payload = create_pull_request_event(
+        action: 'opened',
+        created_at: Time.zone.parse('2020-01-01T15:10:00')
+      )
+
+      create_review_event pull_request_event_payload: pull_request_a_event_payload,
+                          action: 'submitted',
+                          submitted_at: Time.zone.parse('2020-01-01T15:30:00')
+
+      pull_request_b_event_payload = create_pull_request_event(
+        repository_payload: test_repository_b_payload,
+        action: 'opened',
+        created_at: Time.zone.parse('2020-01-01T16:00:00')
+      )
+
+      create_review_event repository_payload: test_repository_b_payload,
+                          pull_request_event_payload: pull_request_b_event_payload,
+                          action: 'submitted',
+                          submitted_at: Time.zone.parse('2020-01-01T16:45:00')
+    end
+
+    it 'it generates the metric for the first project' do
+      expect(first_metric).to have_attributes(
+        entity_key: 'Project A',
+        metric_key: 'review_turnaround',
+        value_timestamp: time_interval_to_process.starting_at
+      )
+
+      expect(first_metric_value_expressed_as_seconds).to be_within(1.second) .of(20.minutes)
+    end
+
+    it 'it generates the metric for the second project' do
+      expect(second_metric).to have_attributes(
+        entity_key: 'Project B',
+        metric_key: 'review_turnaround',
+        value_timestamp: time_interval_to_process.starting_at
+      )
+
+      expect(second_metric_value_expressed_as_seconds).to be_within(1.second) .of(45.minutes)
```

Comment:
what do you think of freezing time instead of working with `be_within`?

Rails comes with helpers by default https://andycroll.com/ruby/replace-timecop-with-rails-time-helpers-in-rspec/

Reply:
or why is `be_within(1.second)` being used?

Reply:
Changed it. Thanks!